Allow installing `yarn` via `corepack` on Debian-based systems, too

hoechenberger opened the pull request:
The default behavior remains unchanged, but Debian and Ubuntu users can now opt-out from the APT-based installation to get the latest `yarn` via `corepack`.

Closes #1004

samruddhikhandale commented on src/node/devcontainer-feature.json:
```diff
@@ -36,6 +36,11 @@
+        "yarnFromApt": {
+            "type": "boolean",
+            "default": true,
+            "description": "Whether to install yarn globally via APT on Debian-based systems (Debian, Ubuntu). If false, will set up yarn via corepack. This setting is ignored on other Linux images, where yarn is always installed via corepack."
```
```suggestion
            "description": "On Debian & Ubuntu systems, you can choose whether to install Yarn globally via APT. If this option is set to false, Yarn will be set up via Corepack instead. This setting does not apply to other Linux distributions, where Yarn is always installed using Corepack."
```

samruddhikhandale commented on src/node/devcontainer-feature.json:
```diff
@@ -36,6 +36,11 @@
             "default": "latest",
             "description": "Version of NVM to install."
+        },
+        "yarnFromApt": {
```
```suggestion
        "installYarnUsingApt": {
```

samruddhikhandale commented on src/node/install.sh:
```diff
@@ -11,6 +11,7 @@ export NODE_VERSION="${VERSION:-"lts"}"
 export NVM_VERSION="${NVMVERSION:-"latest"}"
 export NVM_DIR="${NVMINSTALLPATH:-"/usr/local/share/nvm"}"
 INSTALL_TOOLS_FOR_NODE_GYP="${NODEGYPDEPENDENCIES:-true}"
+export YARN_FROM_APT="${YARNFROMAPT:-true}"  # only concerns Debian-based systems
```
Need to update post rename the Feature option.

samruddhikhandale commented on src/node/install.sh:
```diff
@@ -188,7 +189,7 @@ find_version_from_git_tags() {
 
 install_yarn() {
-    if [ "${ADJUSTED_ID}" = "debian" ]; then
+    if [ "${ADJUSTED_ID}" = "debian" ] && [ "${YARN_FROM_APT}" = "true" ]; then
```
Can we add a test to validate these changes? Thanks!

samruddhikhandale reviewed (changes requested):
This makes sense, thanks! 

Left couple of comments

hoechenberger commented:
@samruddhikhandale I believe I've addressed all your comments. Unfortunately I currently cannot run the test case locally, so we'll have to rely on the CI output

hoechenberger commented on test/node/debian_yarn_from_corepack.sh:
```diff
@@ -0,0 +1,20 @@
+# Definition specific tests
+YARN_VERSION="4.3.0"
+
+# Corepack provides shims for package managers like yarn. When yarn is invoked by the "yarn"
```
```suggestion
# Corepack provides shims for package managers like yarn. When yarn is invoked via the "yarn"
```

samruddhikhandale reviewed (approved):
Looks good, thanks!
